tests/components/knx/test_binary_sensor.py

```diff
@@ -166,21 +162,17 @@ async def test_binary_sensor_counter(hass: HomeAssistant, knx: KNXTestKit) -> No
 
     # receive initial ON telegram
     await knx.receive_write("2/2/2", True)
-    await hass.async_block_till_done()
     # no change yet - still in 1 sec context (additional async_block_till_done needed for time change)
     assert len(events) == 0
     state = hass.states.get("binary_sensor.test")
     assert state.state is STATE_OFF
     assert state.attributes.get("counter") == 0
     async_fire_time_changed(hass, dt_util.utcnow() + timedelta(seconds=context_timeout))
```

Comment:
Should we look into migrating this to the freezer in the future?

Reply:
I can give it a try. Have not worked much with freezer, will need to check if it works here since the lib needs to advance time. But it probably will.

Reply:
addressed in #123038 at 19fe4f03bdab265d1470a623077156de5796e3c1